### PR TITLE
Provide labels to buildings in map #169

### LINKED ISSUE #169: Provide Labels to Buildings in Map

### PR DESCRIPTION
**Pull request by MKDEVELOPEMENT**
in BuildingNames branch I have fixed #169, by creating a button that toggles the display of the names of the buildings

### CODE REVIEW

File: PowerUp/app/src/main/java/powerup/systers/com/MapActivity.java [hunk 1]
```diff
@@ -6,6 +6,8 @@
 import android.view.View;
 import android.view.View.OnClickListener;
 import android.widget.Button;
+import android.widget.TextView;
+
```
Comment (chhavip) [inline PowerUp/app/src/main/java/powerup/systers/com/MapActivity.java]: remove extra line

File: PowerUp/app/src/main/java/powerup/systers/com/MapActivity.java [hunk 2]
```diff
@@ -71,4 +73,41 @@ public DatabaseHandler getmDbHandler() {
     public void setmDbHandler(DatabaseHandler mDbHandler) {
         this.mDbHandler = mDbHandler;
     }
+
```
Comment (chhavip) [inline PowerUp/app/src/main/java/powerup/systers/com/MapActivity.java]: remove extra line

File: PowerUp/app/src/main/java/powerup/systers/com/MapActivity.java [hunk 3]
```diff
@@ -71,4 +73,41 @@ public DatabaseHandler getmDbHandler() {
...
+    public void showText(View v) {
+
+        final TextView houset = (TextView) findViewById(R.id.houset);
+        final TextView hospitalt = (TextView) findViewById(R.id.hospitalt);
+        final TextView schoolt = (TextView) findViewById(R.id.schoolt);
+        final TextView boyfriendt = (TextView) findViewById(R.id.boyfriendt);
+        final TextView storet = (TextView) findViewById(R.id.storet);
+
+        if (text == true) {
+
```
Comment (chhavip) [inline PowerUp/app/src/main/java/powerup/systers/com/MapActivity.java]: remove extra line

File: PowerUp/app/src/main/java/powerup/systers/com/MapActivity.java [hunk 4]
```diff
@@ -71,4 +73,41 @@ public DatabaseHandler getmDbHandler() {
...
+            text = false;
+
```
Comment (chhavip) [inline PowerUp/app/src/main/java/powerup/systers/com/MapActivity.java]: remove extra line

File: PowerUp/app/src/main/java/powerup/systers/com/MapActivity.java [hunk 5]
```diff
@@ -71,4 +73,41 @@ public DatabaseHandler getmDbHandler() {
...
+            houset.setVisibility(View.VISIBLE);
+            hospitalt.setVisibility(View.VISIBLE);
+            schoolt.setVisibility(View.VISIBLE);
+            boyfriendt.setVisibility(View.VISIBLE);
+            storet.setVisibility(View.VISIBLE);
+
+        } else {
+
+            text = true;
+
```
Comment (chhavip) [inline PowerUp/app/src/main/java/powerup/systers/com/MapActivity.java]: remove extra line

File: PowerUp/app/src/main/java/powerup/systers/com/MapActivity.java [hunk 6]
```diff
@@ -71,4 +73,41 @@ public DatabaseHandler getmDbHandler() {
...
+            houset.setVisibility(View.INVISIBLE);
+            hospitalt.setVisibility(View.INVISIBLE);
+            schoolt.setVisibility(View.INVISIBLE);
+            boyfriendt.setVisibility(View.INVISIBLE);
+            storet.setVisibility(View.INVISIBLE);
+
+        }
+
```
Comment (chhavip) [inline PowerUp/app/src/main/java/powerup/systers/com/MapActivity.java]: remove all extra lines

File: PowerUp/app/src/main/res/layout/gamemap.xml [hunk 7]
```diff
@@ -15,6 +15,18 @@
         android:alpha="@dimen/map_alpha"
         android:text="@string/house" />
 
+    <TextView
+        android:layout_width="wrap_content"
```
Comment (chhavip) [inline PowerUp/app/src/main/res/layout/gamemap.xml]: id should be at the top

File: PowerUp/app/src/main/res/layout/gamemap.xml [hunk 8]
```diff
@@ -15,6 +15,18 @@
...
+        android:layout_height="wrap_content"
+        android:layout_below="@id/HouseButton"
+        android:layout_alignLeft="@id/HouseButton"
+        android:layout_alignRight="@id/HouseButton"
+        android:gravity="center"
+        android:text="House"
```
Comment (chhavip) [inline PowerUp/app/src/main/res/layout/gamemap.xml]: This should not be hardcoded but in strings.xml

File: PowerUp/app/src/main/res/layout/gamemap.xml [hunk 9]
```diff
@@ -25,6 +37,18 @@
         android:alpha="@dimen/map_alpha"
         android:text="@string/hospital" />
 
+    <TextView
+        android:layout_width="wrap_content"
```
Comment (chhavip) [inline PowerUp/app/src/main/res/layout/gamemap.xml]: Same changes as before

File: PowerUp/app/src/main/res/layout/gamemap.xml [hunk 10]
```diff
@@ -36,6 +60,19 @@
         android:alpha="@dimen/map_alpha"
         android:text="@string/school" />
 
+    <TextView
+        android:layout_width="wrap_content"
```
Comment (chhavip) [inline PowerUp/app/src/main/res/layout/gamemap.xml]: Same changes as before

File: PowerUp/app/src/main/res/layout/gamemap.xml [hunk 11]
```diff
@@ -46,6 +83,16 @@
         android:alpha="@dimen/map_alpha"
         android:text="@string/boyfriend" />
 
+    <TextView
+        android:layout_width="wrap_content"
```
Comment (chhavip) [inline PowerUp/app/src/main/res/layout/gamemap.xml]: Same changes as before

File: PowerUp/app/src/main/res/layout/gamemap.xml [hunk 12]
```diff
@@ -55,6 +102,18 @@
         android:alpha="@dimen/map_alpha"
         android:text="@string/store" />
 
+    <TextView
+        android:layout_width="wrap_content"
```
Comment (chhavip) [inline PowerUp/app/src/main/res/layout/gamemap.xml]: Same changes as before

File: PowerUp/app/src/main/res/layout/gamemap.xml [hunk 13]
```diff
@@ -64,4 +123,12 @@
         android:layout_alignParentTop="true"
         android:alpha="@dimen/map_alpha" />
 
+    <Button
+        android:layout_width="50dp"
```
Comment (chhavip) [inline PowerUp/app/src/main/res/layout/gamemap.xml]: provide an id to the given button and try to not hard-code the dimensions but strore them in dimens.xml file

**Comment by chhavip** [comment]
There should be only one commit per feature change so also please squash your commits into one. Follow https://github.com/chhavip/Git-Guide for the same and ask if you need more help!

**Comment by MKDEVELOPEMENT** [comment]
@chhavip I have read the document on squashing commits, but I am confused as to where to run these commands

**Comment by chhavip** [comment]
@MKDEVELOPEMENT its fine. make the indicated changes for now, we'll see about squashing  later.

File: PowerUp/app/src/main/java/powerup/systers/com/MapActivity.java (hunk 2 above)
Comment (MKDEVELOPEMENT) [inline PowerUp/app/src/main/java/powerup/systers/com/MapActivity.java]: between which two lines of code?

**Comment by MKDEVELOPEMENT** [comment]
@chhavip I have changed the files, however I am still trying to figure out how to squash commits, from what ive gathered, it is only possible on desktop app? but I am using website. correct me if I'm wrong

**Comment by anubhakushwaha** [comment]
@MKDEVELOPEMENT Go to the command prompt and change the directories till you reach yor project folder.From there follow the steps given in the guide provided starting with cloning the project by git clone command .Further try this interactive site if you want to learn these powerful git commands.[site_link](https://try.github.io/levels/1/challenges/1)

**Comment by MKDEVELOPEMENT** [comment]
how do you change directories in command prompt. and I'm not sure how my command prompt will be linked to my GitHub? since it is a website that I'm using

**Comment by chhavip** [comment]
Closing to look at after #GCI.

**Comment by anubhakushwaha** [comment]
@MKDEVELOPEMENT Have you missed updating the dimen.xml as it's flagging an error for a missing tag.

**Pull request closed without merging**